examples/Consumer/Program.cs

```diff
@@ -132,74 +46,49 @@ public static void Run_Consume(string brokerList, List<string> topics)
                 { "enable.auto.commit", false },
                 { "statistics.interval.ms", 60000 },
                 { "session.timeout.ms", 6000 },
-                { "auto.offset.reset", "smallest" }
+                { "auto.offset.reset", "smallest" },
+                // note: typically, you should treat error_cb events as information only.
+                { "error_cb", (Action<Error>)(error => Console.WriteLine("Error: " + error.Reason)) },
+                { "stats_cb", (Action<string>)(json => Console.WriteLine($"Statistics: {json}")) }
             };
 
             using (var consumer = new Consumer<Ignore, string>(config, null, new StringDeserializer(Encoding.UTF8)))
             {
-                // Note: All event handlers are called on the main thread.
-
-                consumer.OnPartitionEOF += (_, end)
-                    => Console.WriteLine($"Reached end of topic {end.Topic} partition {end.Partition}, next message will be at offset {end.Offset}");
-
-                consumer.OnError += (_, error)
-                    => Console.WriteLine($"Error: {error}");
-
-                // Raised on deserialization errors or when a consumed message has an error != NoError.
-                consumer.OnConsumeError += (_, error)
-                    => Console.WriteLine($"Consume error: {error}");
-
-                // Raised when the consumer is assigned a new set of partitions.
-                consumer.OnPartitionsAssigned += (_, partitions) =>
-                {
-                    Console.WriteLine($"Assigned partitions: [{string.Join(", ", partitions)}], member id: {consumer.MemberId}");
-                    // If you don't add a handler to the OnPartitionsAssigned event,
-                    // the below .Assign call happens automatically. If you do, you
-                    // must call .Assign explicitly in order for the consumer to 
-                    // start consuming messages.
-                    consumer.Assign(partitions);
-                };
+                // Note: All event handlers are called on the main .Consume thread.
+                
+                // Raised when the consumer has been notified of a new assignment set.
```

Comment:
yes, good idea to document this here.

In the version you reviewed, I reverted these event names to `OnPartitionsAssigned/Revoked` (I agree, I didn't like the received terminology I came up with). but this choice of naming is offending me too much. I've now changed them to `OnPartitionAssignment` and `OnPartitionAssignmentRevoked`. I like these names - they clearly describe what's going on rather than conflating it.

Reply:
They use two different tenses:
OnPartitionAssignment is a noun while ..Revoked is a verb.

Is there any actual community confusion from the existing names, if not we should not change it.

Prior art, Java: https://docs.confluent.io/current/clients/javadocs/org/apache/kafka/clients/consumer/ConsumerRebalanceListener.html#onPartitionsAssigned-java.util.Collection-

Reply:
In java: "This method will be called after the partition re-assignment completes", so `onPartitionsAssigned` is a good name - it happens after partitions have been assigned. In C#, it happens after the assignment notification, but before the assign - it's a bad, misleading name. There is plenty of indication that people don't get the assignment process properly in general. Yes I take your point on the naming. I will keep thinking.